includes/class-sensei-frontend.php

```diff
@@ -835,7 +835,7 @@ public function sensei_frontend_messages() {
 	 * @param int $post_id Optional. Lesson ID. Default 0.
 	 */
 	public function sensei_lesson_video( $post_id = 0 ) {
-		if ( 0 < intval( $post_id ) && sensei_can_user_view_lesson( $post_id ) ) {
```

Comment:
I get the point that it seems unnecessary but this is legacy code and it may have been added for a reason back then. Is there any other reason you are removing it, maybe a performance issue or something else?

Reply:
Yeah, lesson videos are only viewable for enrolled users. We can break out the code getting the video from the meta into a new method and use that in `get_featured_video_html`, and keep this one for the hook. That new method can also return the HTML instead of echoing, so the output buffering wrapper can be left out.

Reply:
Thanks @onubrooks and @yscik - I will address this!